server/consumer.go

```diff
@@ -2534,19 +2538,18 @@ func (o *consumer) applyState(state *ConsumerState) {
 		return
 	}
 
-	// If o.sseq is greater don't update. Don't go backwards on o.sseq.
-	if o.sseq <= state.Delivered.Stream {
+	// If o.sseq is greater don't update. Don't go backwards on o.sseq if leader.
+	if !o.isLeader() || o.sseq <= state.Delivered.Stream {
```

Comment:
```suggestion
	if o.isFollower() || o.sseq <= state.Delivered.Stream {
```

Is there a value in adding more Domain-specific language?

Reply:
Should be noted that there's no such function as `(*consumer).isFollower()` right now. IMO "not leader" is clear enough.

Reply:
Yeah, the intent was to suggest to continue expanding the Domain-Specific code, instead of relying on Symbols.